section/authentication.html

```
@@ -0,0 +1,302 @@
+  </section>
+
+  <section id="keypair-generation">
+    <h3>Generating an Ed25519 key pair</h3>
+    <p>
+      This algorithm creates a fresh Ed25519 key pair and persists it
+      as {{Storage/keys}} in {{Storage}}.
+    </p>
+    <div class="algorithm">
+      To <dfn data-lt="generate an Ed25519 key pair|produce HTTP signatures">generate an Ed25519 key pair</dfn>:
+      <ol>
+        <li>Let |keyId:DOMString| be a newly generated UUID v4 ([[RFC9562]]).</li>
+        <li>Generate a new [=Ed25519 private key=] and corresponding [=Ed25519 public key=] per [[RFC8032]].</li>
+        <li>Let |rawPublicKeyBytes| be the raw public key bytes.</li>
+        <li>Let |publicKeyJwk:DOMString| be the result of running the [=construct a public key JWK=] algorithm
+          given |rawPublicKeyBytes| and |keyId|.</li>
+        <li>Let |privateKeySeed:PrivateKeySeed| be the 32-octet seed bytes.</li>
+        <li>
+          Let |authKeys:AuthKeys| be a new {{AuthKeys}} instance with the following members:
+          <ul>
+            <li>{{AuthKeys/keyId}} field set to |keyId|</li>
+            <li>{{AuthKeys/privateKey}} field set to |privateKeySeed|</li>
+            <li>{{AuthKeys/publicKey}} field set to |publicKeyJwk|</li>
+          </ul>
+        </li>
+        <li>
+          Run the [=store auth keys=] algorithm with |authKeys|, overwriting
+          any existing value.
+        </li>
+      </ol>
+    </div>
+  </section>
+
+  <section>
+    <h3>HTTP message signatures</h3>
+    <p>User agents MUST create HTTP message signatures for authenticated Open Payments requests using the Ed25519 key pair. The following algorithm defines signature construction.</p>
+    <div class="algorithm">
+      To <dfn>generate an HTTP message signature</dfn>, given
+      an HTTP <var data-type="request">request</var>:
+      <ol>
+        <li>Let |authKeys:AuthKeys?| be the result of the [=get auth keys=] algorithm.</li>
+        <li>If |authKeys| is null, then return.</li>
```

Can we say "fail" or something instead of "return"?